File: 은지/LV2/가장 큰 정사각형 찾기.js
function solution(board)
{
    var answer = 0;
    const n = board.length
    const m = board[0].length

    if (n === 1 && m === 1 && board[0][0] === 1) {
      return 1
    }

    for (let i=1; i<n; i++) {
      for (let j=1; j<m; j++) {
        if (board[i][j]) {
          board[i][j] = Math.min(board[i-1][j-1], board[i][j-1], board[i-1][j]) + 1
          if (answer < board[i][j]) {
            answer = board[i][j]
          }
        }
      }
    }

    return answer ** 2;
}

File: 은지/LV2/가장 큰 정사각형 찾기.py
def solution(board):
    answer = 0
    n = len(board) # 행 개수
    m = len(board[0]) # 열 개수

    if n == 1 and m == 1 and board[0][0] == 1:
        return 1

    for i in range(1, n):
        for j in range(1, m):
            if board[i][j]:
                board[i][j] = min(board[i-1][j-1], board[i-1][j], board[i][j-1])+1
                if answer < board[i][j]:
                    answer = board[i][j]
    # print(board) 
    return answer ** 2

b = [[1]]
print(solution(b))